src/dbnode/storage/series/buffer.go

```diff
@@ -114,13 +114,55 @@ type bufferStats struct {
 
 type bufferTickResult struct {
 	mergedOutOfOrderBlocks int
-	evictedBuckets         int
+	evictedBucketTimes     times
+}
+
+// times is a struct that holds an unknown number of times. This is used to
+// to avoid heap allocations as much as possible by trying to not allocate a
+// slice of times. To do this, `timesArraySize` needs to be strategically sized
+// such that for the vast majority of the time, the internal array can hold all
+// the times required so that `slice` is nil.
+type times struct {
+	arrIdx int
+	arr    [timesArraySize]time.Time
```

Comment:
considering you're doing this to be incredibly space conscious, would using xtime.UnixNano instead of time.Time be better here?